internal/flag/internal_flag.go

```diff
@@ -340,6 +355,20 @@ func (f *InternalFlag) GetVariationValue(name string) interface{} {
 	return nil
 }
 
+func (f *InternalFlag) GetTargetingKey(ctx ffcontext.Context) (string, error) {
```

Comment:
I will rename this function `GetBucketingKey` because this is why we are using it, we want the field used for the bucketing.
```suggestion
func (f *InternalFlag) GetBucketingKey(ctx ffcontext.Context) (string, error) {
```